examples/Mixtures/slot_based.py

```diff
@@ -0,0 +1,278 @@
+
+parameters = [
+    slot1_label,
+    slot2_label,
+    slot3_label,
+    slot1_amount,
+    slot2_amount,
+    slot3_amount,
+]
+
+### Constraint Setup
+
+# For the sake of demonstration, we consider a scenario where we do *not* care about the
+# order of addition of components to the mixture, which imposes two additional
+# constraints.
+#
+# ```{admonition} Order of Addition
+# :class: note
+# Whether you need to impose the constraints for removing duplicates and imposing
+# permutation invariance depends on your use case. If the order of addition is relevant
+# to your mixture, the permutation invariance constraint should be discarded and one
+# could further argue that adding the same substance multiple times should be allowed.
+# ```
+
+#### Duplicate Substances
+
+# Assuming that the order of addition is irrelevant, there is no difference between
+# having two slots with the same substance or having only one slot with the combined
+# amounts. Thus, we want to make sure that there are no such duplicate label entries:
+
+no_duplicates_constraint = DiscreteNoLabelDuplicatesConstraint(
+    parameters=["Slot1_Label", "Slot2_Label", "Slot3_Label"]
+)
+
+#### Permutation Invariance
+
+# Next, we need to take care of permutation invariance. If our order of addition does
+# not matter, the result of interchanging any two slots does not alter the overall
+# mixture, i.e. the mixture slots are are considered permutation-invariant.
+
+# A complication with permutation invariance arises from the fact that we have not
+# only a label per slot, but also a numerical amount. If this amount is zero, then the
+# label of the slot becomes meaningless (i.e. which ingredient should be considered for the
+# slot), because adding zero of it does not change the mixture. In BayBE, we call
+# this a "dependency", i.e. the slot labels depend on the slot amounts and are only
+# relevant if the amount satisfies some condition (in this case "amount > 0").
+
+# The {class}`~baybe.constraints.discrete.DiscreteDependenciesConstraint` informs the
+# {class}`~baybe.constraints.discrete.DiscretePermutationInvarianceConstraint` about
+# these dependencies so that they are correctly included in the filtering process:
+
+perm_inv_constraint = DiscretePermutationInvarianceConstraint(
+    parameters=["Slot1_Label", "Slot2_Label", "Slot3_Label"],
+    dependencies=DiscreteDependenciesConstraint(
+        parameters=["Slot1_Amount", "Slot2_Amount", "Slot3_Amount"],
+        conditions=[
+            ThresholdCondition(threshold=0.0, operator=">"),
+            ThresholdCondition(threshold=0.0, operator=">"),
+            ThresholdCondition(threshold=0.0, operator=">"),
+        ],
+        affected_parameters=[["Slot1_Label"], ["Slot2_Label"], ["Slot3_Label"]],
+    ),
+)
+
+#### Substance Amounts
+
+# Interpreting the slot amounts as percentages, we need to ensure that their total is
+# always 100:
+
+sum_constraint = DiscreteSumConstraint(
```

Comment:
Shouldn't we actually rather promote the `from_simplex` constructor here?

Reply:
Agree. So I suggest: let's keep the code as is but add an admonition informing people that the shown version is for educational purposes only and that the simplex constructor offers an optimized solution. Would you be happy with that?

Reply:
Have added the admonition. If you like it, simply resolve